### DISCUSSION TITLE
Cache Duplication/Racing (GHA)

### DISCUSSION THREAD

**Discussion by nolan-yoo-miris** [microsoft/vcpkg#42216] [Q&A]
Hi - I'm part of a team managing a monorepo building for multiple platforms, and we're at a bit of a loss.

We're using vcpkg's binary caching with GitHub Actions, and we very quickly hit the per-repo 10GB limit. (We are currently on Enterprise Cloud - not Server, so can't increase this via policy.)

This wasn't a surprise at first glance - our cache for one platform can hit 5GB - but taking a closer look, the behavior looks a bit unexpected?

![image](https://github.com/user-attachments/assets/9df15cf0-5a55-4a70-8de3-cc083fde5c62)

We appear to be "re-caching" some of the binaries, which doesn't seem desired.

Are we doing something wrong? Is there a way to fix this?

Thank you

**Comment by dg0yt** [comment]
Are you aware that the default triplets build both debug and release configurations?

**Reply by nolan-yoo-miris** [reply]
This is possibly our issue - we were originally using custom triplets but I'll check with my team

- Does it not suffice to set `VCPKG_BUILD_TYPE`?
- Will debug vs release produce different ABI hashes?
